### PR TITLE
Fixed report of unnecessary parentheses when assigning a lambda to a val

### PR DESCRIPTION
**Pull request by schalkms**
reference #767

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt [hunk 1]
```diff
@@ -42,7 +43,8 @@ import org.jetbrains.kotlin.psi.KtValueArgument
 class UnnecessaryParentheses(config: Config = Config.empty) : Rule(config) {
 
 	override val issue = Issue("UnnecessaryParentheses", Severity.Style,
-			"Unnecessary parentheses don't add any value to the code and should be removed.")
+			"Unnecessary parentheses don't add any value to the code and should be removed.",
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt]: Probably even less than 5mins.

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt]: That's the minimum detekt currently has.

File: docs/pages/documentation/style.md [hunk 2]
```diff
@@ -630,7 +630,8 @@ class B : Object()
 
 ### UnnecessaryParentheses
 
-Reports unnecessary parentheses around expressions.
+This rule reports unnecessary parentheses around expressions.
+This unnecessary parentheses can safely be removed.
```
Comment (vanniktech) [inline docs/pages/documentation/style.md]: Should be: These parentheses

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParentheses.kt]: when you count git add/commit and referencing an issue, it should be min. 5 minutes :D

**Pull request merged**